build-livecd-root

```diff
@@ -1,95 +1,102 @@
-  exit 1
+	echo "Please run this script as root"
+	exit 1
 fi
 
-which livecd-creator isohybrid implantisomd5 >/dev/null || ( echo "Command(s) missing, install required tools" && exit 2 )
-
-trap cleanup EXIT
+which livemedia-creator pykickstart >/dev/null || ( echo "Command(s) missing, install required tools" && exit 2 )
 
 srcdir=$(readlink -f $(dirname $0))
 destdir=${2:-$srcdir}
-tmpdir=$(mktemp -d /tmp/build-fdi-XXXXXX)
+kernelcmd=${3:-nomodeset nokaslr}
+tmpdir=$(mktemp -d ${FDI_BUILD_DIR:-/var/tmp}/fdi-XXXXXXXXX)
 
-cd $srcdir
+pushd $srcdir
 last_tag=${1:-$(git describe --abbrev=0 --tags)}
 last_sha=$(git log --pretty=format:'%h' -n 1)
+popd
 
-cd $tmpdir
-echo Working in directory $tmpdir
+if [ ! -e install.iso ]; then
+       echo "* Downloading netboot installation ISO"
+       wget -O install.iso http://isoredirect.centos.org/centos/8-stream/isos/x86_64/CentOS-Stream-8-x86_64-latest-boot.iso
+fi
 
-echo "* Running livecd-creator"
-livecd-creator -v --title="Discovery Image" --compression-type=xz --cache /var/cache/build-fdi --config $srcdir/fdi-image.ks -f fdi -t /tmp
+echo "* Building ISO (kernel cmdline: $kernelcmd)"
+# https://bugzilla.redhat.com/show_bug.cgi?id=1955836 - dmsquash-live-ntfs
+# https://bugzilla.redhat.com/show_bug.cgi?id=2034601 - product CentOS Linux workaround
+livemedia-creator --ks fdi-image.ks \
+	--make-iso \
+	--iso install.iso \
+	--nomacboot \
+	--extra-boot-args "$kernelcmd" \
+	--project "FDI" \
+	--releasever "$last_tag/$last_sha" \
+	--anaconda-arg="--product CentOS Linux" \
+	--dracut-arg="--xz" \
+	--dracut-arg="--no-hostonly" \
+	--dracut-arg="--debug" \
+	--dracut-arg="--no-early-microcode" \
+       	--dracut-arg="--omit plymouth" \
+        --dracut-arg="--add livenet dmsquash-live convertfs pollcdrom qemu qemu-net" \
+        --dracut-arg="--add-drivers mptbase mptscsih mptspi hv_storvsc hid_hyperv hv_netvsc hv_vmbus" \
+	--tmp "$tmpdir"
 
 if [ $? -ne 0 ]; then
-    KEEP_TMPDIR=yes
-    echo "Error creating livecd"
+    echo "Error creating livecd, use KEEP_TMPDIR to investigate"
     exit 1
 fi
 
-echo "* Remastering ISO with correct kernel options"
-bash -x $srcdir/aux/remaster/discovery-remaster fdi.iso fdi.local fdi-bootable.iso
-
-echo "* Filtering ISO with isohybrid"
-isohybrid --partok --uefi fdi-bootable.iso
-
-echo "* Implanting MD5 sum"
-implantisomd5 --force fdi-bootable.iso
-mv fdi-bootable.iso $destdir/fdi-bootable-${last_tag#release-}.iso
+# Move manually, --resultdir lorax option does not work correctly on CentOS8
+echo "* Moving the ISO to the destination directory"
+DEST_ISO="$destdir/fdi-$last_tag-$last_sha.iso"
+mv -f $tmpdir/lmc-*/images/boot.iso "$DEST_ISO"
+chown --reference $srcdir/build-livecd "$DEST_ISO"
 
 echo "* Converting to initrd"
-bash -x $srcdir/aux/livecd-iso-to-pxeboot $tmpdir/fdi.iso
+aux/livecd-iso-to-pxeboot "$DEST_ISO"
+rm -rf tftpboot/pxelinux.{cfg,0}
+mv tftpboot/initrd.img tftpboot/initrd0.img
+mv tftpboot/vmlinuz tftpboot/vmlinuz0
+chmod 666 tftpboot/*
```

Comment:
Does it need to be world writable?

Reply:
Nope 644 is enough, foreman-proxy is the owner and TFTP server needs to be able to read. Fixed.